tests/transforms/test_qmc_transform.py

```diff
@@ -26,10 +26,54 @@
     quantum_monte_carlo,
 )
 from pennylane.templates.subroutines.qmc import _make_V, _make_Z, make_Q
-from pennylane.templates.state_preparations.mottonen import _uniform_rotation_dagger as r_unitary
 from pennylane.wires import Wires
 
 
+def r_unitary(gate, alpha, control_wires, target_wire):
+    r"""Applies a uniformly-controlled rotation to the target qubit.
+
+    A uniformly-controlled rotation is a sequence of multi-controlled
+    rotations, each of which is conditioned on the control qubits being in a different state.
+
+    For example, a uniformly-controlled rotation with two control qubits describes a sequence of
+    four multi-controlled rotations, each applying the rotation only if the control qubits
+    are in states :math:`|00\rangle`, :math:`|01\rangle`, :math:`|10\rangle`, and :math:`|11\rangle`, respectively.
+
+    To implement a uniformly-controlled rotation using single qubit rotations and CNOT gates,
+    a decomposition based on Gray codes is used. For this purpose, the multi-controlled rotation
+    angles alpha have to be converted into a set of non-controlled rotation angles theta.
+    For more details, see `Möttönen and Vartiainen (2005), Fig 7a<https://arxiv.org/pdf/quant-ph/0504100.pdf>`_.
+
+    Args:
+        gate (.Operation): gate to be applied, needs to have exactly one parameter
+        alpha (tensor_like): angles to decompose the uniformly-controlled rotation into multi-controlled rotations
+        control_wires (array[int]): wires that act as control
+        target_wire (int): wire that acts as target
+    """
+
+    theta = qml.templates.state_preparations.mottonen._compute_theta(alpha)
```

Comment:
It seems a bit weird to call a private function from another module... might make sense to make `compute_theta` public?